roles/ovirt-image-template/tasks/main.yml

```diff
@@ -1,23 +1,31 @@
 ---
+- name: source of the disk to import
+  debug:
+    var: source
+
 - name: Download the qcow image
   get_url:
     url: "{{ qcow_url }}"
     dest: "{{ image_path }}"
   register: downloaded
+  when: source  == "qcow_file"
```

Comment:
Please separate it into two files:
- qcow2_image.yml
- glance_image.yml

and import it based on params, something like:

```yaml
- name: Fail when invalid parameters
  fail:
    msg: "You must either specify qcow2_url or glance_image"
  when: "glance_image is defined and qcow2_url is defined"

- name: Glance image
  include: glance_image.yml
  when: "glance_image is defined"

- name: QCOW2 image
  include: qcow2_image.yml
  when: "qcow2_url is defined"
```